tasks/release.py

```diff
@@ -831,3 +840,61 @@ def create_rc(
         security_agent_policies_version,
         macos_build_version,
     )
+
+    # Update internal module dependencies
+    update_modules(ctx, _stringify_version(highest_version))
+
+
+def check_version(agent_version):
+    """Check Agent version to see if it is valid."""
+    version_re = re.compile(r'7[.](\d+)[.](\d+)(-rc\.(\d+))?')
+    if not version_re.match(agent_version):
+        raise Exit(message="Version should be of the form 7.Y.Z or 7.Y.Z-rc.t")
+
+
+@task
+def update_modules(ctx, agent_version, verify=True):
+    """
+    Update internal dependencies between the different Agent modules.
+    * --verify checks for correctness on the Agent Version (on by default).
+
+    Examples:
+    inv -e release.update-modules 7.27.0 
+    """
+    if verify:
+        check_version(agent_version)
+
+    for module in DEFAULT_MODULES.values():
+        for dependency in module.dependencies:
+            dependency_mod = DEFAULT_MODULES[dependency]
+            ctx.run(
+                "go mod edit -require={dependency_path} {go_mod_path}".format(
+                    dependency_path=dependency_mod.dependency_path(agent_version), go_mod_path=module.go_mod_path()
+                )
+            )
+
+
+@task
+def tag_version(ctx, agent_version, commit="HEAD", verify=True, push=True):
+    """
+    Create tags for a given Datadog Agent version.
+    The version should be given as an Agent 7 version.
+
+    * --commit COMMIT will tag COMMIT with the tags (default HEAD)
+    * --verify checks for correctness on the Agent version (on by default).
+    * --push will push the tags to the origin remote (on by default).
```

Comment:
Should this be on by default? I think it's fine (the worst that can happen is an artifact on the staging repo) but we can have it off by default.